src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java

```diff
@@ -146,4 +147,49 @@ public void testSaveAndLoadMessages() throws PersistentDataStoreException {
     Assert.assertEquals(contentTwo, resultMessageTwo.getContent());
     Assert.assertEquals(creationTwo, resultMessageTwo.getCreationTime());
   }
+
+    @Test
+    public void testSaveAndLoadActivities() throws PersistentDataStoreException {
+        UUID idOne = UUID.fromString("10000000-2222-3333-4444-555555555555");
+        String descriptionOne = "User1 has joined!";
+        Instant datetimeOne = Instant.ofEpochMilli(1);
+        int typeOne = 0;
+        Activity activityOne =
+                new Activity(idOne, descriptionOne, datetimeOne, typeOne);
+
+        UUID idTwo = UUID.fromString("10000000-2222-3333-4444-555555555556");
+        String descriptionTwo = "User2 has joined!";
+        Instant datetimeTwo = Instant.ofEpochMilli(2);
+        int typeTwo = 0;
+        Activity activityTwo =
+                new Activity(idTwo, descriptionTwo, datetimeTwo, typeTwo);
+
+        // save
+        persistentDataStore.writeThrough(activityOne);
+        persistentDataStore.writeThrough(activityTwo);
+
+        // load since ofEpochMilli(3) excluded
+        List<Activity> expectTwoResults = persistentDataStore.loadActivitiesBeforeDatetime(Instant.ofEpochMilli(3), 10);
```

Comment:
milli 2 and milli 1 are exclusively earlier than milli 3, and so we expected them both to be retrieved